File: Difficulty: Easy/Reverse First K elements of Queue/reverse-first-k-elements-of-queue.py
from collections import deque

class Solution:
    def modifyQueue(self, q, k):
        # Initialize an empty list to use as a stack
        s = []

        # Pop the first k elements from the queue and push them into the stack
        for _ in range(k):
            s.append(q.popleft())

        # Pop all elements from the stack and push them back into the queue
        while s:
            q.append(s.pop())

        # Now, move the remaining elements from the front of the queue to the back
        # to preserve their order
        for _ in range(len(q) - k):
            q.append(q.popleft())

        return q


#{ 
 # Driver Code Starts
#Initial Template for Python 3
import atexit
import io
import sys
from collections import deque

#Contributed by : Nagendra Jha

_INPUT_LINES = sys.stdin.read().splitlines()
input = iter(_INPUT_LINES).__next__
_OUTPUT_BUFFER = io.StringIO()
sys.stdout = _OUTPUT_BUFFER

@atexit.register

def write():
    sys.__stdout__.write(_OUTPUT_BUFFER.getvalue())

if __name__ == '__main__':
    test_cases = int(input())
    for cases in range(test_cases) :
        n,k = map(int,input().strip().split())
        a = list(map(int,input().strip().split()))

        queue = deque() # our queue to be used
        for i in range(n):
            queue.append(a[i]) # enqueue elements of array in our queue
        
        ob = Solution()
        result = ob.modifyQueue(queue,k)
        list_result = []
        while result:
            list_result.append(result.popleft())
        print(*list_result)
        
        print("~")
# } Driver Code Ends

File: Difficulty: Easy/Reverse First K elements of Queue/reverse-first-k-elements-of-queue.cpp
//{ Driver Code Starts
// Initial Template for C++

#include <bits/stdc++.h>
using namespace std;

vector<int> inputLine() {
    string str;
    getline(cin, str);
    stringstream ss(str);
    int num;
    vector<int> res;
    while (ss >> num) {
        res.push_back(num);
    }
    return res;
}


// } Driver Code Ends
//Back-end complete function Template for C++

class Solution
{
    public:
    // Function to reverse first k elements of a queue.
    queue<int> modifyQueue(queue<int> q, int k) {
        // Create a stack to store the first k elements
        stack<int> St;

        // Step 1: Pop the first k elements from the queue and push them to the stack
        for(int i = 0; i < k; i++) {
            int temp = q.front();
            q.pop();
            St.push(temp);
        }

        // Step 2: Pop from the stack and push into the queue to reverse the order
        for(int i = 0; i < k; i++) {
            int temp = St.top();
            St.pop();
            q.push(temp);
        }

        // Step 3: Move the remaining elements (size - k) to the back of the queue
        int m = q.size() - k;
        for(int i = 0; i < m; i++) {
            int temp = q.front();
            q.pop();
            q.push(temp);
        }

        // Return the modified queue
        return q;
    }
};



//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    cin.ignore();
    while (t-- > 0) {
        int k;
        cin >> k;
        cin.ignore();
        queue<int> q;
        vector<int> v = inputLine();
        for (auto it : v)
            q.push(it);
        Solution ob;
        queue<int> ans = ob.modifyQueue(q, k);
        while (!ans.empty()) {
            int a = ans.front();
            ans.pop();
            cout << a << " ";
        }
        cout << endl;
        cout << "~" << endl;
    }
}
// } Driver Code Ends